### DISCUSSION TITLE
mu_initial_guess

### DISCUSSION THREAD

**Discussion by KeyuAN95** [TRIQS/solid_dmft#91] [Q&A]
Dear Alex and all,

I run the solid_dmft with the following dmft_config.toml:
#######################################
[general]
seedname = "test_wannier"
jobname = "U3.1"
beta = 40
n_iw = 251
n_tau = 5001
n_iter_dmft = 10
prec_mu = 0.0001
mu_initial_guess = 13.3115
h_int_type = "kanamori"
U = 3.1
J = 0.0

dc_type = 0
dc_dmft = false
h5_save_freq = 1
[solver]
type = "cthyb"
length_cycle = 60
n_warmup_cycles = 1e+4
n_cycles_tot = 2e+6
imag_threshold = 1e-6
measure_density_matrix = true
delta_interface=true
perform_tail_fit = true
####################################
it |         mu |   G(beta/2) per orbital |    orbital occs up+down |      impurity occ
  0 |    0.55267 |   -0.04555     -0.06437 |    0.98401      0.97000 |           1.95402
  1 |    0.55267 |   -0.04519     -0.06516 |    1.00095      0.95128 |           1.95223
  2 |    0.54320 |   -0.04623     -0.06262 |    0.97786      0.95671 |           1.93457
  3 |    0.54569 |   -0.04363     -0.06560 |    0.96887      0.94197 |           1.91084
  4 |    0.55155 |   -0.04418     -0.06334 |    0.98258      0.95281 |           1.93539
  5 |    0.56119 |   -0.04419     -0.06583 |    0.96832      0.94570 |           1.91402
  6 |    0.56700 |   -0.04438     -0.06588 |    0.96992      0.95199 |           1.92191
  7 |    0.57212 |   -0.04386     -0.06572 |    0.99978      0.98294 |           1.98273
  8 |    0.56013 |   -0.04249     -0.06476 |    0.99534      0.94508 |           1.94041
  9 |    0.56597 |   -0.04259     -0.06269 |    0.97671      0.96611 |           1.94282
 10 |    0.56296 |   -0.04551     -0.06560 |    0.99362      0.97021 |           1.96383
 11 |    0.55860 |   -0.04308     -0.06911 |    0.97626      0.95802 |           1.93428
 12 |    0.54908 |   -0.04250     -0.06567 |    0.99067      0.96105 |           1.95172
 13 |    0.54729 |   -0.04468     -0.06503 |    0.98642      0.94993 |           1.93635
 14 |    0.54440 |   -0.04401     -0.06558 |    0.97823      0.95619 |           1.93442
 15 |    0.54615 |   -0.04377     -0.06541 |    0.98066      0.94751 |           1.92818
 16 |    0.54934 |   -0.04522     -0.06455 |    0.97793      0.95845 |           1.93638
 17 |    0.55049 |   -0.04417     -0.06407 |    0.97252      0.95499 |           1.92751
 18 |    0.55487 |   -0.04380     -0.06579 |    0.96830      0.96988 |           1.93818
 19 |    0.55017 |   -0.04273     -0.06703 |    0.96811      0.94844 |           1.91655
 20 |    0.55858 |   -0.04452     -0.06304 |    0.96412      0.94878 |           1.91291
 21 |    0.55739 |   -0.04497     -0.06566 |    0.98887      0.96901 |           1.95788
 22 |    0.55346 |   -0.04397     -0.06506 |    0.98979      0.95225 |           1.94204
 23 |    0.55253 |   -0.04305     -0.06582 |    0.97203      0.94926 |           1.92129
 24 |    0.55842 |   -0.04342     -0.06513 |    0.97468      0.96106 |           1.93574
 25 |    0.56121 |   -0.04355     -0.06468 |    0.98758      0.94158 |           1.92915
 26 |    0.56668 |   -0.04331     -0.06546 |    0.98298      0.94715 |           1.93013
 27 |    0.56503 |   -0.04498     -0.06393 |    0.98962      0.93672 |           1.92634
 28 |    0.56664 |   -0.04342     -0.06412 |    1.00418      0.95557 |           1.95975
 29 |    0.56685 |   -0.04350     -0.06405 |    0.98903      0.94795 |           1.93698
 30 |    0.56777 |   -0.04442     -0.06585 |    0.98341      0.96469 |           1.94809
##################################################################################
After the running, the chemical potential μ did not converge to a fixed value. How can I deal with this?

**Comment by KeyuAN95** [comment]
I can achieve convergence of the chemical potential mu to a fixed value by removing the Fermi energy in .inp.

**Reply by the-hampel** [reply]
Hi @KeyuAN95 , 

sorry for the slow response. Somehow I missed your post. Yes probably the chemical potential was not found correctly, since you specified an initial guess (13.xx) and the actual fermi level was now around 0.0. As you correctly noticed this is because adding it in the .inp file will subtract it when converting to the h5 format for solid_dmft. 

Best,
Alex
